### PR TITLE
Update supported Elixir, OTP and Ecto versions

### PR DESCRIPTION
**Pull request by remi**
## 📖 Description and reason

This pull request brings changes from #34 and #36 as well as a few changes to make sure everything still works for all supported versions.

@simonprev I’m going to need your 👍 on this one since I had to change a few things to make tests pass.

This should close #34 and #36. Thank you @kianmeng & @martinmaillard!

## 🦀 Dispatch

`#dispatch/elixir`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,25 +4,27 @@ on: [push, pull_request]
...
+    name: Erlang ${{matrix.otp-version}} / Elixir ${{matrix.elixir-version}}
+    runs-on: ubuntu-20.04
     strategy:
       matrix:
-        elixir-version: [1.12.x, 1.11.x, 1.10.x, 1.9.x, 1.8.x]
+        # https://hexdocs.pm/elixir/compatibility-and-deprecations.html
+        elixir-version: [1.17.x, 1.16.x, 1.15.x, 1.14.x, 1.13.x]
         include:
-          - elixir-version: 1.12.x
+          - elixir-version: 1.17.x
```
Comment (kianmeng) [inline .github/workflows/ci.yml:14]: We should also include OTP 26 and 27 as well?

Comment (remi) [reply .github/workflows/ci.yml:14]: Yep absolutely, I just wanted to make all OTP 25 builds go green and then check for 26 and 27 after 🙂

Comment (remi) [reply .github/workflows/ci.yml:13]: @kianmeng I added a few!

```yaml
- elixir-version: 1.17.x
  otp-version: 27.x
- elixir-version: 1.17.x
  otp-version: 26.x
- elixir-version: 1.16.x
  otp-version: 26.x
- elixir-version: 1.15.x
  otp-version: 26.x
```

🙌

**Review by simonprev — APPROVED** [review]

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**